tritium-registry/src/main/java/com/palantir/tritium/metrics/registry/RealMetricName.java

```diff
@@ -90,6 +90,10 @@ static MetricName create(MetricName other, String extraTagName, String extraTagV
         return new RealMetricName(other.safeName(), withEntry(other.safeTags(), extraTagName, extraTagValue));
     }
 
+    public static RealMetricName create(MetricName metricName, TagMap tag) {
+        return new RealMetricName(metricName.safeName(), tag.withEntries(metricName.safeTags()));
```

Comment:
need to think through how we want to handle overlap for tags as this may be the inverse of what we'd want